### PR TITLE
Update snapshot-restore.md

### PR DESCRIPTION
**Pull request by pjuri**
### Description
Documentation update. Helps user to avoid "Connect timed out" error when registering a custom S3 endpoint for snapshots.

### Issues Resolved
Closes #16305
https://github.com/opensearch-project/OpenSearch/issues/16305

### Version
All

### Frontend features
n/a

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md [hunk 1]
```diff
@@ -110,6 +110,20 @@ You will most likely not need to specify any parameters except for `location`. F
    sudo ./bin/opensearch-keystore add s3.client.default.secret_key
    ```
 
+1. (Optional) If you're using custom S3 endpoint (e.g. MinIO), disable AWS EC2 metadata connection:
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:113]:
```suggestion
1. (Optional) If you're using a custom S3 endpoint (for example, MinIO), disable AWS EC2 metadata connection:
```

File: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md [hunk 2]
```diff
@@ -110,6 +110,20 @@ You will most likely not need to specify any parameters except for `location`. F
...
+
+   ```bash
+   export AWS_EC2_METADATA_DISABLED=true
+   ```
+
+   If you're installing OpenSearch via Helm, put this in your values file:
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:119]: Does this 
```suggestion
   If you're installing OpenSearch using Helm, append the following to your values file:
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @pjuri! A couple of suggestions.

File: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md (hunk 2 above)
Comment (pjuri) [reply _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:119]: You're not appending, you're changing an existing parameter with an empty value (extraEnvs: []): https://artifacthub.io/packages/helm/opensearch-project-helm-charts/opensearch?modal=values

Comment (kolchfa-aws) [reply _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:119]: Got it. I updated the suggestion.

File: _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md [hunk 3]
```diff
@@ -110,6 +110,20 @@ You will most likely not need to specify any parameters except for `location`. F
    sudo ./bin/opensearch-keystore add s3.client.default.secret_key
    ```
 
+1. (Optional) If you're using a custom S3 endpoint (for example, MinIO), disable AWS EC2 metadata connection:
+
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/snapshots/snapshot-restore.md:114]: Line 113: The service is named "Amazon EC2" (never "AWS EC2"). Additionally, if these are the first appearances of these service names, they should be "Amazon Simple Storage Service (Amazon S3) endpoint" and "Amazon Elastic Compute Cloud (Amazon EC2) metadata connection". If these are not the first appearances, "S3 endpoint" and "EC2 metadata connection" are fine.

**Review by natebower — COMMENTED** [review]
@pjuri @kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

**Pull request merged**